lib/api/model/initial_snapshot.dart

```diff
@@ -660,12 +666,115 @@ class SupportedPermissionSettings {
 @JsonSerializable(fieldRename: FieldRename.snake)
 class PermissionSettingsItem {
   final bool allowEveryoneGroup;
+
+  @DefaultGroupNameConverter()
+  final SettableAsDefaultGroupName defaultGroupName;
+
   // also other fields not yet used
 
-  PermissionSettingsItem({required this.allowEveryoneGroup});
+  PermissionSettingsItem({
+    required this.allowEveryoneGroup,
+    required this.defaultGroupName,
+  });
 
   factory PermissionSettingsItem.fromJson(Map<String, dynamic> json) =>
     _$PermissionSettingsItemFromJson(json);
 
   Map<String, dynamic> toJson() => _$PermissionSettingsItemToJson(this);
 }
+
+/// A value of [PermissionSettingsItem.defaultGroupName].
+///
+/// Can be any of these:
+/// - a known system group [SystemGroupName]
+/// - a known special string [PseudoSystemGroupName]
+/// - an unknown system group or special string [DefaultGroupNameUnknown]
+sealed class SettableAsDefaultGroupName {
+  String toJson();
+}
+
+class DefaultGroupNameConverter extends JsonConverter<SettableAsDefaultGroupName, String> {
+  const DefaultGroupNameConverter();
+
+  @override
+  SettableAsDefaultGroupName fromJson(String json) {
+    final SettableAsDefaultGroupName? maybeResult = json.startsWith('role:')
+      ? SystemGroupName.fromJson(json)
+      : PseudoSystemGroupName.fromJson(json);
+    return maybeResult ?? DefaultGroupNameUnknown(json);
+  }
+
+  @override
+  String toJson(SettableAsDefaultGroupName object) => object.toJson();
+}
+
+class DefaultGroupNameUnknown extends SettableAsDefaultGroupName {
+  DefaultGroupNameUnknown(this.apiValue);
+  final String apiValue;
+
+  @override
+  String toJson() => apiValue;
+}
+
+/// A known special string
+/// that [PermissionSettingsItem.defaultGroupName] might be.
+///
+/// See server implementation, e.g.
+/// `can_administer_channel_group` in zerver/models/streams.py.
+@JsonEnum(valueField: 'apiValue', alwaysCreate: true)
+enum PseudoSystemGroupName implements SettableAsDefaultGroupName {
+  streamCreatorOrNobody(apiValue: 'stream_creator_or_nobody'),
```

Comment:
On CZO I proposed documenting this: [#api design > stream_creator_or_nobody @ 💬](https://chat.zulip.org/#narrow/channel/378-api-design/topic/stream_creator_or_nobody/near/2258637)

From that discussion, it looks like it might also get renamed.